neural_modelling/src/synapse_expander/connection_generators/connection_generator_fixed_pre.h

```diff
@@ -79,13 +91,110 @@ static void *connection_generator_fixed_pre_initialise(address_t *region) {
 
     // Initialise the RNG
     obj->rng = rng_init(region);
-    log_debug("Fixed Number Pre Connector,  pre_lo = %u, pre_hi = %u, "
-    		"post_lo = %u, post_hi = %u, allow self connections = %u, "
+    log_debug("Fixed Number Pre Connector parameters: pre_lo = %u, pre_hi = %u, "
+            "post_lo = %u, post_hi = %u, allow self connections = %u, "
             "with replacement = %u, n_pre = %u, n pre neurons = %u",
-			obj->params.pre_lo, obj->params.pre_hi, obj->params.post_lo, obj->params.post_hi,
+            obj->params.pre_lo, obj->params.pre_hi, obj->params.post_lo, obj->params.post_hi,
             obj->params.allow_self_connections,
             obj->params.with_replacement, obj->params.n_pre,
             obj->params.n_pre_neurons);
+
+    // Build the array
+
+    // Get how many values can be sampled from
+    uint32_t n_values = obj->params.n_pre_neurons;
+
+    // Get the number of connections in each column
+    uint32_t n_conns = obj->params.n_pre;
+
+    log_debug("Generating %u from %u possible synapses", n_conns, n_values);
+
+    // The number of columns is the number of post-slices to do the calculation for
+    uint32_t n_columns = obj->params.post_hi - obj->params.post_lo + 1;
+
+    // Allocate array for each column (i.e. post-slice on this slice)
+    uint16_t (*array)[n_columns][n_conns] =
+            spin1_malloc(n_columns * n_conns * sizeof(uint16_t));
+    fixed_pre_globals.in_sdram = false;
+    if (array == NULL) {
+        log_warning("Could not allocate in DTCM, trying SDRAM");
+        array = sark_xalloc(sv->sdram_heap,
+                n_columns * n_conns * sizeof(uint16_t), 0, ALLOC_LOCK);
+        fixed_pre_globals.in_sdram = true;
+    }
+    if (array == NULL) {
+        log_error("Could not allocate array for indices");
+        rt_error(RTE_SWERR);
+    }
+
+    // Loop over the columns and fill the full_indices array accordingly
+    for (uint32_t n = 0; n < n_columns; n++) {
```

Comment:
From my (limited) investigations of this, I think because each core generates the same thing each time, then this shouldn't be a problem... but yes, as you say, that could be done in a different PR if necessary.

Reply:
It will work, but what is generated differs depending on the post_vertex_slice.  That said, the keys currently depend on that anyway, so yes, I think this can be left for now...